src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -183,5 +190,20 @@ void FindBundleHeader()
 
             return headerOffset != 0;
         }
+
+        private static void SetFilePermission(string path)
+        {
+           if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+           {
+                var retCode = chmod(path, filePermissionOctal);
+                if (retCode != 0)
+                {
+                    throw new Win32Exception(retCode, $"Could not set file permission {filePermissionOctal} for {path}.");
```

Comment:
On failure `chmod` returns `-1` and `errno` is set.
`errno` can be retrieved using `Marshal.GetLastWin32Error()`.
So you want to pass that as the first argument to `Win32Exception` instead of `retCode`.